lib/compress.js

```diff
@@ -1731,6 +1731,7 @@ merge(Compressor.prototype, {
                 var val = {};
                 for (var i = 0, len = this.properties.length; i < len; i++) {
                     var prop = this.properties[i];
+                    if (prop instanceof AST_Expansion) return this;
```

Comment:
`drop_side_effect_free()` turns out to be already covered by `def(AST_Expansion)`:
https://github.com/mishoo/UglifyJS2/blob/4700c14855e0189804e0834bc1a12e39c4463961/lib/compress.js#L2884-L2886

As for `may_throw_on_access`:
```
$ echo 'var a = {get b(){throw 0}}; ({...a}).b;' | node
[stdin]:1
var a = {get b(){throw 0}}; ({...a}).b;
                              ^^^

SyntaxError: Unexpected token ...
    at createScript (vm.js:74:10)

$ echo 'var a = {get b(){throw 0}}; ({...a}).b;' | bin/uglifyjs -c | node

$ echo 'var a = {get b(){throw 0}}; ({...a}).b;' | bin/uglifyjs -c
var a={get b(){throw 0}};
```
Note the property access thus exception gets dropped.